repositories/migrations/20240521171800_add_to_case_workflow_indexes.sql

```diff
@@ -0,0 +1,26 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE INDEX decisions_add_to_case_idx ON decisions (org_id, pivot_value, case_id)
+WHERE
+      pivot_value IS NOT NULL;
+
+CREATE INDEX cases_add_to_case_workflow_idx ON cases (org_id, inbox_id, id)
+WHERE
+      status IN ('open', 'investigating');
+
+DROP INDEX decisions_case_id_idx;
+
+CREATE INDEX decisions_case_id_idx ON decisions (org_id, case_id) INCLUDE (pivot_value);
+
+-- +goose StatementEnd
+-- +goose Down
+-- +goose StatementBegin
+DROP INDEX decisions_case_id_idx;
+
+DROP INDEX cases_add_to_case_workflow_idx;
+
+DROP INDEX decisions_add_to_case_idx;
+
+CREATE INDEX decisions_case_id_idx ON decisions (org_id, case_id);
+
+-- +goose StatementEnd
```

Comment:
FYI, I found an existing index defined like this in our migration files:
```sql
CREATE INDEX decisions_pivot_value_index ON decisions (org_id, pivot_value, created_at DESC);
```

Reply:
Yea I know, it's close enough. Maybe it would have been sufficient, at least in most cases. The thing is, to provide for the case where there are many entries in both sides of the join (which can happen), I wanted to make sure we do  a merge join rather than a loop join, for which I want the decisions to be filtered by org_id/pivot_value and ordered by case_id, which is what I added this index for. I'll make sure to monitor this once it's live